### ISSUE TITLE
Conflict between 2 vendor folders during deploy

### ISSUE THREAD

**Issue by florentdestremau** [composer/composer#6983]
I am having a very weird bug while testing out a new symfony 3.4 flex project. When I tried to deploy it on a dummy server, the first deploy went well, but then on the second this happened... And I have no idea about what to do.

My `composer.json`:

```json
{
    "type": "project",
    "license": "proprietary",
    "require": {
        "php": "^7.0.8",
        "deployer/deployer": "^6.0",
        "symfony/console": "^3.4",
        "symfony/flex": "^1.0",
        "symfony/form": "^3.4",
        "symfony/framework-bundle": "^3.4",
        "symfony/lts": "^3",
        "symfony/maker-bundle": "^1.0",
        "symfony/orm-pack": "^1.0",
        "symfony/profiler-pack": "^1.0",
        "symfony/security-bundle": "^3.4",
        "symfony/twig-bundle": "^3.4",
        "symfony/validator": "^3.4",
        "symfony/yaml": "^3.4"
    },
    "require-dev": {
        "symfony/dotenv": "^3.4"
    },
    "config": {
        "preferred-install": {
            "*": "dist"
        },
        "sort-packages": true
    },
    "autoload": {
        "psr-4": {
            "App\\": "src/"
        }
    },
    "autoload-dev": {
        "psr-4": {
            "App\\Tests\\": "tests/"
        }
    },
    "scripts": {
        "auto-scripts": {
            "cache:clear": "symfony-cmd",
            "assets:install --symlink --relative %PUBLIC_DIR%": "symfony-cmd"
        },
        "post-install-cmd": [
            "@auto-scripts"
        ],
        "post-update-cmd": [
            "@auto-scripts"
        ]
    },
    "conflict": {
        "symfony/symfony": "*"
    },
    "extra": {
        "symfony": {
            "id": "01C31HWYZBEKDJGV29RXC1NH18",
            "allow-contrib": false
        }
    }
}
```

Output of `composer diagnose`:

```
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.6.2
PHP version: 7.1.12-3+ubuntu16.04.1+deb.sury.org+1
PHP binary path: /usr/bin/php7.1
```

When I run this command (note the folder: `releases/3`):

```
export APP_ENV='prod'; cd ~/symfony-docker/releases/3 && /usr/bin/php /usr/local/bin/composer install --verbose --prefer-dist --no-progress --no-interaction --no-dev --optimize-autoloader

```

I get the following output (note that the conflict is with a *different* folder, `releases/2`:

```
[......]
[159.89.31.105] <  Extracting archive
[159.89.31.105] <   - Installing symfony/validator (v3.4.2):
[159.89.31.105] < Loading from cache
[159.89.31.105] <  Extracting archive
[159.89.31.105] < Generating optimized autoload files
[159.89.31.105] < ocramius/package-versions:  Generating version class...
[159.89.31.105] < ocramius/package-versions: ...done generating version class
[159.89.31.105] < > post-install-cmd: @auto-scripts
[159.89.31.105] < Executing script cache:clear
[159.89.31.105] < Executed script cache:clear  [KO]
[159.89.31.105] <  [KO]
[159.89.31.105] < Script cache:clear returned with error code 255
[159.89.31.105] < !!  PHP Fatal error:  Cannot declare interface Symfony\Component\HttpKernel\HttpKernelInterface, because the name is already in use in /root/symfony-docker/releases/2/vendor/symfony/http-kernel/HttpKernelInterface.php on line 22

```

And I expected this to happen:

`Deploy done :) `

Other info: i used a server with dply.co so a digital ocean Ubuntu 16.04.
My php install locally (I then run with docker) is as follows

```
sudo apt-get install -y python-software-properties
sudo add-apt-repository -y ppa:ondrej/php
sudo apt-get update
sudo apt-get install -y php7.1 php7.1-mbstring php7.1-curl php7.1-mbstring php7.1-mcrypt php7.1-zip php7.1-int php7.1-bcmath php7.1-xml php7.1-opcache
curl -sS https://getcomposer.org/installer | sudo php -- --install-dir=/usr/local/bin --filename=composer
```

I tried setting `php -d apc.enable_cli=0` as the php binary, didn't work. 
I tried force-setting opcache.enable_cli=0 in my `php.ini` but didn't work too.
I use deployer.org to deploy.

Would like any help to understand this.

**Comment by xabbuh** [comment]
This looks like the same as #6977. However, that should have been resolved in Composer 1.6.2 which you seem to be using. Could it be that the other directory was set up when installing dependencies with Composer 1.6.0. or 1.6.1?

**Comment by florentdestremau** [comment]
It was just a dummy project to test out some stuff, so no it was a brand new deploy, and same composer version, update whatsoever between the deploys, within the same hour or so.
I tried deleting the whole files and 1st deploy always works, but second doesn't.

**Comment by Seldaek** [comment]
@xabbuh #6977 was *not* finding classes. In this case the class is found but it exists already so it fatals.

**Comment by florentdestremau** [comment]
@Seldaek I will try to reproduce this this evening, with different composer versions. Is there an easy way to do this ?

**Comment by Seldaek** [comment]
@florentdestremau you can run `composer self-update 1.6.2` or any other version to get a specific one.

**Comment by florentdestremau** [comment]
couldn't reproduce. Don't know what it was but it's solved now :thinking: , with same version

**Comment by xabbuh** [comment]
Did you update the Symfony version you use?

**Comment by florentdestremau** [comment]
Mmmh no not really. I guess I should, if someone has the same problem they can still re-open, right ?

**Comment by cooldude77** [comment]
My two cents
So this is happening to my installation too. I have slightly modified php deployer tool and it was working till Symfony- 3.4 php - 5.6.40
I installed my app on php >=7.2.. ( symfony unchanged ) using deployer and just after one deployment, I got the problem of HttpKernelnterface. 
So I went ahead and deleted shared/var/cache and did cache clear as shown in the pic. 
I redeployed and it worked. 

I think working of bin/console has changed ( probably for php  = 7.2) which is causing this issue.  

![Screenshot_26](https://user-images.githubusercontent.com/2662881/56950972-432da300-6b54-11e9-94e0-d6f6f7dbf45f.png)
Hope this helps.
